client/src/index.tsx

```diff
@@ -11,6 +12,10 @@ const router = createBrowserRouter([
     path: '/',
     element: <App />,
   },
+  {
+    path: '*',
+    element: <PageNotFound />,
+  },
```

Comment:
```suggestion
const router = createBrowserRouter([
  {
    path: '/',
    element: <App />,
    errorElement: <PageNotFound />,
  },
  {
    path: '/post',
    element: <PostCard post={post} />,
  },
]);
```
and move the router to a separate file